pkg/controller/secrets/dns.go

```diff
@@ -0,0 +1,99 @@
+package secrets
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/acorn-io/baaah/pkg/router"
+	"github.com/acorn-io/runtime/pkg/config"
+	"github.com/acorn-io/runtime/pkg/labels"
+	"github.com/acorn-io/runtime/pkg/publish"
+	"github.com/acorn-io/runtime/pkg/system"
+	"golang.org/x/exp/slices"
+	corev1 "k8s.io/api/core/v1"
+	netv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func HandleDNSSecret(req router.Request, resp router.Response) error {
+	sec := req.Object.(*corev1.Secret)
+
+	domain := string(sec.Data["domain"])
+	if domain == "" {
+		return fmt.Errorf("acorn dns secret is misconfigured, missing domain")
+	}
+
+	cfg, err := config.Get(req.Ctx, req.Client)
+	if err != nil {
+		return err
+	}
+
+	if strings.EqualFold(*cfg.AcornDNS, "disabled") || !slices.Contains(cfg.ClusterDomains, domain) {
```

Comment:
pretty sure the clusterDomains check covers the case where the feature is disabled, but theres no real harm in checking both

Reply:
Talked offline and updated.